src/renderer/atlas/AtlasEngine.api.cpp

```diff
@@ -401,7 +403,50 @@ void AtlasEngine::ToggleShaderEffects() noexcept
 }
 
 [[nodiscard]] HRESULT AtlasEngine::UpdateFont(const FontInfoDesired& fontInfoDesired, FontInfo& fontInfo, const std::unordered_map<std::wstring_view, uint32_t>& features, const std::unordered_map<std::wstring_view, float>& axes) noexcept
-try
+{
+    static constexpr std::array fallbackFaceNames{ static_cast<const wchar_t*>(nullptr), L"Consolas", L"Lucida Console", L"Courier New" };
+    auto it = fallbackFaceNames.begin();
+    const auto end = fallbackFaceNames.end();
+
+    for (;;)
+    {
+        try
+        {
+            _updateFont(*it, fontInfoDesired, fontInfo, features, axes);
+            return S_OK;
+        }
+        catch (...)
+        {
+            ++it;
+            if (it == end)
+            {
+                RETURN_CAUGHT_EXCEPTION();
+            }
+            else
+            {
+                LOG_CAUGHT_EXCEPTION();
+            }
+        }
+    }
+}
+
+void AtlasEngine::UpdateHyperlinkHoveredId(const uint16_t hoveredId) noexcept
+{
+    _api.hyperlinkHoveredId = hoveredId;
+}
+
+#pragma endregion
+
+void AtlasEngine::_resolveAntialiasingMode() noexcept
+{
+    // If the user asks for ClearType, but also for a transparent background
+    // (which our ClearType shader doesn't simultaneously support)
+    // then we need to sneakily force the renderer to grayscale AA.
+    const auto forceGrayscaleAA = _api.antialiasingMode == D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE && !_api.backgroundOpaqueMixin;
+    _api.realizedAntialiasingMode = forceGrayscaleAA ? D2D1_TEXT_ANTIALIAS_MODE_GRAYSCALE : _api.antialiasingMode;
+}

```

Comment:
This code isn't new. It's just moved up from down below to group the code better.

Reply:
this makes it difficult to review the diff -- can you point out the part that fixed the fallback issue you mentioned?

Reply:
Well, frankly, I didn't move the code.
It's just git diffing it poorly and no builtin algorithm improves this diff. The `_updateFont` function is new.